src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -473,6 +503,77 @@ public void RemoveAdditionalFile(string fullPath)
 
         #endregion
 
+        #region Non Source File Addition/Removal
+        public void AddDynamicSourceFile(string dynamicFilePath, ImmutableArray<string> folders)
+        {
+            var extension = FileNameUtilities.GetExtension(dynamicFilePath)?.TrimStart('.');
+            if (extension?.Length == 0)
+            {
+                return;
+            }
```

Comment:
Why?  What is this for?

Reply:
provider is picked up by extension. if given file doesn't have file extension, we skip those.